### ISSUE TITLE
Where can I find the complete changelog

### ISSUE THREAD

**Issue by 37176427** [spring-projects/spring-data-elasticsearch#2286]
Sorry to ask this question.
After upgrading from 4.3.2 to 4.4.2 ,Many changes have taken place that I did not anticipate.
For example,
1 the default value of UpdateQuery's lang field is changed from "painless" to null,
2 the pageable field of the NativeSearchQuery has changed from UNPAGED (size :10000)to DEFAULT_PAGED(size :10)
So that I need to modify every place where it is used
Due to too many such problems, I cannot safely upgrade to 4.4.2. 

I have searched: 
docs.spring.io/spring-data/elasticsearch/docs/4.4.2/reference/html/#elasticsearch-migration-guide-4.3-4.4
github release & issue

**Comment by sothawo** [comment]
Hello,

> 1 the default value of UpdateQuery's lang field is changed from "painless" to null,

The default language used in Elasticsearch when no value is passed in is `painless`. We changed from explicitly setting the default value of Elasticsearch to implicitly using it by not setting it in the request. For the user this does not change anything.

> 2 the pageable field of the NativeSearchQuery has changed from UNPAGED (size :10000)to DEFAULT_PAGED(size :10)

I see, this was not set on the `NativeSearchQuery` but on the `NativeSearchQueryBuilder`, the default on the `Query` implementations has always been `DEFAULT_PAGED`.  The query builders were refactored when the integration for the new Elasticsearch client was introduced, and in the `BaseQueryBuider` where the different query builder classes derive from, this is set to `null`. This leads again to the default behaviour of Elasticsearch to return chunks of 10 hits.

As for the complete changelog, I think that VMware in the build process generates a changelog from the commits, I have to ask where this is published.

**Comment by 37176427** [comment]
For Q1,If the language is not set before the upgrade, **painless** will be used.
             but after the upgrade, a null pointer exception will be thrown. This is where I found the problem
```
java.lang.NullPointerException: null
        at java.util.Objects.requireNonNull(Objects.java:203)
        at org.elasticsearch.script.Script.<init>(Script.java:540)
        at org.elasticsearch.script.Script.<init>(Script.java:519)
        at org.springframework.data.elasticsearch.core.RequestFactory.updateRequest(RequestFactory.java:1030)
        at org.springframework.data.elasticsearch.core.RequestFactory.lambda$bulkRequest$1(RequestFactory.java:250)
```
For Q2,It is indeed set in the builder's method. Because I didn't explicitly specify paging before, I had to supplement paging settings in all places after upgrading.
Finally, thank you very much for your reply~

**Comment by sothawo** [comment]
The NPE should not happen of course, this is now fixed with #2288.
